src/workerd/io/container-client-test.js

```diff
@@ -0,0 +1,281 @@
+import { DurableObject } from 'cloudflare:workers';
+import assert from 'node:assert';
+
+const CONTAINER_NAME = 'container-example';
+const IMAGE_TAG = 'ubuntu';
+
+export class DurableObjectExample extends DurableObject {
+  async testBasics() {
+    let container = this.ctx.container;
+    assert.strictEqual(container.running, false);
+
+    // Test environment variable validation
+    const badEnvs = [
+      {
+        value: { A: 'B\0' },
+        err: "Environment variable values cannot contain '\\0': A",
+      },
+      {
+        value: { 'B\0': 'A' },
+        err: "Environment variable names cannot contain '\\0': B\0",
+      },
+      {
+        value: { 'B=': 'A' },
+        err: "Environment variable names cannot contain '=': B=",
+      },
+    ];
+
+    for (const badEnv of badEnvs) {
+      try {
+        await container.start({
+          entrypoint: ['/bin/sh'],
+          env: badEnv.value,
+          enableInternet: true,
```

Comment:
I think that's the only we care about here

```suggestion
          env: badEnv.value,
```

Reply:
I prefer keeping it since, we also validate these inputs behavior here.